logdna/resource_view.go

```diff
@@ -14,19 +15,19 @@ type clientViewPayload struct {
 }
 
 type channel struct {
-	BodyTemplate    map[string]string `json:"bodyTemplate,omitempty"`
-	Emails          []string          `json:"emails,omitempty"`
-	Headers         map[string]string `json:"headers,omitempty"`
-	Immediate       string            `json:"immediate,omitempty"`
-	Integration     string            `json:"integration,omitempty"`
-	Key             string            `json:"key,omitempty"`
-	Method          string            `json:"method,omitempty"`
-	Operator        string            `json:"operator,omitempty"`
-	Terminal        string            `json:"terminal,omitempty"`
-	Triggerinterval string            `json:"triggerinterval,omitempty"`
-	Triggerlimit    int               `json:"triggerlimit,omitempty"`
-	Timezone        string            `json:"timezone,omitempty"`
-	URL             string            `json:"url,omitempty"`
+	BodyTemplate    map[string]interface{} `json:"bodyTemplate,omitempty"`
```

Comment:
curious, why is this `map[string]interface{}`? it looks like it's just a string via `jsonencode`. On line 108-109 it gets parsed, but then gets stringified in the request again. I did confirm that test still pass after making this `string` and removing lines 108-109 so just wanted to see if there was another reason for this.

Reply:
I need to use the decoding otherwise I consistently get this error from the config-api which says that bodytemplate is not represented as an object. I've added decode errors as well as renamed the field to bt instead of body template

Reply:
that's fine, from the [test](https://github.com/logdna/terraform-provider-logdna/pull/7/files#diff-a73081a7213e3d706376480aa6657980f9307c5a4fbe3f65ba7301a96c72b1f3R310) it looked like it **was** already being sent to the API as a pre-encoded string.